src/ctapipe/core/component.py

```diff
@@ -233,7 +232,7 @@ def get_current_config(self):
         for val in self.__dict__.values():
             if isinstance(val, Component):
                 config[name].update(val.get_current_config())
-            if isinstance(val, Iterable):
+            if isinstance(val, (list, tuple)) and isinstance(val[0], Component):
```

Comment:
need to perform a length check, this raises an exception in case of empty list / tuple